great_expectations/experimental/datasources/data_asset/data_connector/data_connector.py

```diff
@@ -0,0 +1,190 @@
+    ) -> None:
+        self._name: str = name
+
+        self._datasource_name: str = datasource_name
+        self._data_asset_name: str = data_asset_name
+
+        # This is a dictionary which maps data_references onto batch_requests.
+        self._data_references_cache: Dict = {}
+
+        self._data_context_root_directory: Optional[pathlib.Path] = None
+
+    @property
+    def name(self) -> str:
+        return self._name
+
+    @property
+    def data_asset_name(self) -> str:
+        return self._data_asset_name
+
+    @property
+    def datasource_name(self) -> str:
+        return self._datasource_name
+
+    @property
+    def data_context_root_directory(self) -> Optional[pathlib.Path]:
+        return self._data_context_root_directory
+
+    @data_context_root_directory.setter
+    def data_context_root_directory(
+        self, data_context_root_directory: Optional[pathlib.Path]
+    ) -> None:
+        self._data_context_root_directory = data_context_root_directory
+
+    def build_batch_spec(self, batch_definition: BatchDefinition) -> BatchSpec:
+        """
+        Builds batch_spec from batch_definition by generating batch_spec params and adding any pass_through params
+
+        Args:
+            batch_definition (BatchDefinition): required batch_definition parameter for retrieval
+        Returns:
+            BatchSpec object built from BatchDefinition
+
+        """
+        batch_spec_params: dict = (
+            self._generate_batch_spec_parameters_from_batch_definition(
+                batch_definition=batch_definition
+            )
+        )
+        batch_spec = BatchSpec(**batch_spec_params)
+        return batch_spec
+
+    def get_batch_definition_list_from_batch_request(
+        self, batch_request: BatchRequest
+    ) -> List[BatchDefinition]:
+        raise NotImplementedError
+
+    def get_data_reference_count(self) -> int:
+        raise NotImplementedError
+
+    def get_unmatched_data_references(self) -> List[Any]:
+        raise NotImplementedError
+
+    def _generate_batch_spec_parameters_from_batch_definition(
+        self, batch_definition: BatchDefinition
+    ) -> dict:
+        raise NotImplementedError
+
+    def _refresh_data_references_cache(
+        self,
+    ) -> None:
+        raise NotImplementedError
+
+    def _map_data_reference_to_batch_definition_list(
+        self, data_reference: Any
+    ) -> Optional[List[BatchDefinition]]:
+        raise NotImplementedError
+
+    def _map_batch_definition_to_data_reference(
+        self, batch_definition: BatchDefinition
+    ) -> Any:
+        raise NotImplementedError
+
+    def _get_data_reference_list(self) -> List[str]:
+        """
+        List objects in the underlying data store to create a list of data_references.
+        This method is used to refresh the cache by classes that extend this base DataConnector class
+        """
+        raise NotImplementedError
+
+    def self_check(self, pretty_print=True, max_examples=3):
```

Comment:
this isn't necessary, data connectors are not configured in ZEP

Reply:
@donaldheppner While the name `self_check()` may not be idea (please suggest a better one), this is a nice utility that calls key `DataConnector` interface methods, which makes it convenient in tests.  It can also potentially help us in our ZEP `connection_test()` logic (although I have not yet fully explored the product requirements for this feature).  Thanks.

Reply:
The method introduces requirements in the base class that are otherwise unnecessary, for the purpose of printing, which is also undesirable. This method should be removed

Reply:
@donaldheppner I do not see it introducing requirements -- could you please point me to a requirement you have spotted so that I can investigate?  All I am seeing is a convenience method that provides a combined report for useful aspects of the `DataConnector` behavior, which comes in handy in tests (nothing new is introduced).  Or are you seeing something else?  Happy to discuss.  Thank you.